examples/individual_properties/main.rs

```diff
@@ -0,0 +1,166 @@
+/*
+ * This file is part of Astarte.
+ *
+ * Copyright 2021 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+use std::convert::TryFrom;
+use std::env;
+use std::fs::File;
+use std::io::BufReader;
+
+use nom::{
+    bytes::complete::tag, character::complete::digit1, error::ErrorKind, sequence::delimited,
+};
+use serde::{Deserialize, Serialize};
+
+use astarte_device_sdk::{
+    database::AstarteSqliteDatabase, options::AstarteOptions, types::AstarteType, AstarteDeviceSdk,
+    AstarteError,
+};
+
+#[derive(Serialize, Deserialize, Debug)]
+struct Config {
+    realm: String,
+    device_id: String,
+    credentials_secret: String,
+    pairing_url: String,
+}
+
+// Getter function for the property "name" of a sensor.
+async fn get_name_for_sensor(device: &AstarteDeviceSdk, sensor_n: i32) -> Result<String, String> {
+    let interface = "org.astarte-platform.rust.examples.individual-properties.DeviceProperties";
+    let path = format!("/{sensor_n}/name");
+
+    if let Some(name) = device
+        .get_property(interface, &path)
+        .await
+        .map_err(|e| e.to_string())?
+    {
+        if let AstarteType::String(name) = name {
+            Ok(name)
+        } else {
+            Err("Incorrect property type".to_string())
+        }
+    } else {
+        Ok("None".to_string())
+    }
+}
+
+#[tokio::main]
+async fn main() -> Result<(), AstarteError> {
+    env_logger::init();
+
+    // Load configuration
+    let file = File::open("./examples/individual_properties/configuration.json").unwrap();
+    let cfg: Config = serde_json::from_reader(BufReader::new(file)).unwrap();
+
+    // Open the database, create it if it does not exists
+    let this_file = env::current_dir()
+        .unwrap()
+        .join("examples/individual_properties/astarte-example-db.sqlite");
+    let db = AstarteSqliteDatabase::new(this_file.to_str().unwrap()).await?;
```

Comment:
```suggestion
    let db = AstarteSqliteDatabase::new("./examples/individual_properties/astarte-example-db.sqlite").await?;
```

Reply:
This could be also improved by saving the database to a temporary directory with `tempdir` that would be cleaned when the example finishes